File: hyeonsook95/baekjoon/10025.ts
const fs = require('fs');
const input = fs.readFileSync('dev/stdin').toString().trim().split('\n');

const [N, K] = input[0].split(' ').map(num => parseInt(num));
var maps = Array(1000001).fill(0);

for (var inputIdx = 0; inputIdx < N; inputIdx++) {
    var [ice, idx] = input[inputIdx + 1].trim().split(' ').map(num => parseInt(num));
    maps[idx] = ice;
}

var result = 0;
var end = (2 * K + 1) > 1000001 ? 1000001 : (2 * K + 1);
for (var idx = 0; idx < end; idx++) {
    result += maps[idx];
}

var left = 0;
var right = 2 * K;
var pSum = result;
while (right < 1000000) {
    left += 1;
    right += 1;

    pSum = pSum - maps[left - 1] + maps[right];
    result = pSum > result ? pSum : result;
}

console.log(result);

File: hyeonsook95/baekjoon/10025.py
import sys

input = sys.stdin.readline

maps = [0] * 1000001

N, K = map(int, input().strip().split(" "))

for _ in range(N):
    ice, idx = map(int, input().strip().split(" "))
    maps[idx] = ice

result = sum(maps[: 2 * K])
p_sum = result
left, right = 0, 2 * K
while right < 1000000:
    left += 1
    right += 1

    p_sum = p_sum - maps[left - 1] + maps[right]
    result = max(result, p_sum)


print(result)
